src/test/java/com/ea/restaurant/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,87 @@
+package com.ea.restaurant.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.User;
+import com.ea.restaurant.repository.UserRepository;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+class UserServiceImplTest {
+  @MockBean private UserRepository userRepository;
```

Comment:
You don't need `@MockBean` here since this is a unit test. Please just mock the `UserRepository`

Reply:
Done.